Bottom: 4c6d2e9df781cbbbeec34480eb893058adba32a0
Top:    f3fba68a281f61b2b8d127dd067f497b8bfd27d2
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-04 15:39:51 +0000

Workaround crash where contour is passed as NULL to poly_PreContour










---

diff --git a/src/polygon.c b/src/polygon.c
index 43432bc..7410329 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -247,6 +247,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
